src/navigator_data_ingest/base/new_document_actions.py

```diff
@@ -113,12 +114,32 @@ def _handle_document(
     _LOGGER.info(f"Handling document: {document}")
 
     session = requests.Session()
+
+    try:
+        document_source_url = (
+            pydantic.AnyHttpUrl(document.source_url) if document.source_url else None
+        )
```

Comment:
This is quite specific, could be worth adding a comment or updating the docstring to give some brief context

Reply:
Added comments to the docstring